### PR TITLE
minikube: Add the kvm2 driver

### PR DESCRIPTION
**Pull request by shosti**
###### Motivation for this change

The docker-machine-kvm driver is considered deprecated should be replaced with
the docker-machine-kvm2 driver (which is maintained by the minikube team).

See NixOS/nixpkgs#34448

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/cluster/minikube/default.nix [hunk 1]
```diff
@@ -1,9 +1,31 @@
...
 let
+  version = "0.26.0";
+  src = fetchFromGitHub {
+    owner  = "kubernetes";
+    repo   = "minikube";
+    rev    = "v${version}";
+    sha256 = "1wc2gvmgb59yh0ldm2plvh6s8mvxvysrxp6w75z16ii86jmi3wr6";
+  };
+
+  docker-machine-kvm2 = buildGoPackage rec {
```
Comment (nlewo) [inline pkgs/applications/networking/cluster/minikube/default.nix]: Could you move out this expression and put it close to the `docker-machine-kvm` attribute?

**Review by nlewo — CHANGES_REQUESTED** [review]
Does minikube work fine with this driver?

**Comment by shosti** [comment]
Hmm it worked for me building on 18.03, will check again in a bit...

**Comment by flokli** [comment]
unstable here, with sandbox enabled.

**Comment by nlewo** [comment]
The git dependency fails to build for me.
@GrahamcOfBorg build minikube

**Comment by nlewo** [comment]
Hmm, that strange because for me and Hydra a dependency of minikube cannot be build:
```
$ git log -n1 --oneline 
c24a7f1ada6 (HEAD -> pr-39438) minikube: Add the kvm2 driver

$ nix-instantiate -A minikube.all | tail -n1 | xargs nix-store -qR | grep kubernetes
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
/nix/store/ap9adnrvrmd04fvl3ia5is2z4x8nrbsr-kubernetes-1.9.1.drv
```
This derivation is failing on Hydra https://hydra.nixos.org/build/73254958#tabs-summary.

Can you rebase onto master?

(note I don't know which nixpkgs commit ofborg is checking out...)

**Comment by akazakov** [comment]
@flokli I got the same error, which I was able to get around by providing the version explicitly 
`minikube start --kubernetes-version v1.10.0 --vm-driver kvm2`

All seems to work after that 
```
Starting local Kubernetes v1.10.0 cluster...
Starting VM...
Getting VM IP address...
Moving files into cluster...
Downloading kubeadm v1.10.0
Downloading kubelet v1.10.0
Finished Downloading kubeadm v1.10.0
Finished Downloading kubelet v1.10.0
Setting up certs...
Connecting to cluster...
Setting up kubeconfig...
Starting cluster components...
Kubectl is now configured to use the cluster.
Loading cached images from config file.
```

**Comment by flokli** [comment]
@nlewo I did rebase on latest nixos-unstable (channels/nixos-unstable), minikube evaluated to `/nix/store/k2zs1s8sr6sc4c0fh5dik5pgsvpq3kfh-minikube-0.26.0-bin`.

@akazakov Passing the version manually works for me. Can you check if passing it automatically works if we patch `hack/get_k8s_version.py` to the version we're trying to build?

**Comment by flokli** [comment]
superseded by #41644 which got merged. @nlewo, close?

**Pull request closed without merging**